AENS.md

```diff
@@ -347,6 +376,30 @@ the name enters the `revoked` state. After a fixed timeout of
 2016 blocks, the name will be available for claiming again.
 
 
+### Subname
+
+```
+ ------------ ----
+| name       | 63 |
+ ------------ ----
+| definition |    |
+ ------------ ----
+```
+
+The `subname` transaction MUST be signed by the owner of the `name` entry.
+
+`subname` transaction is used to define a tree of subnames below `name` entry
+along with the pointers for each subname.
+
+A `definition` field is a mapping of `subname` to `pointers`, where the `pointers`
```

Comment:
If no subname is meant to be without parent, then this needs to be detailed for preventing attacks. I am focusing on the semantics layer - not on the deserialization layer.

A. If constraint on parenthood enforced strictly, then malicious user may post subname tx filled with subnames up to microblock gas limit, and having as last name an orphan subname. What is behaviour in this case? If dropping all tx (hence not charging sender), this does not account for the computational cost of the past large portion of the list of subnames. Recommended solution: add a SHOULD constraint on subnames being a tree, noting that subnames that are orphan are silently ignored (the idea is so that user gets charged for computational cost).

B. If tx needs to be processed fully in order to know whether subname shall be added or not, this opens the door for tx computational cost non-linear with size of tx. Recommended solution: add a SHOULD constraint on subnames being sorted according to a specific algorithm you specify here (TO-BE-DONE) for making computational cost of check more predictable.

C. Can subnames have arbitrary depth? Please comment on max depth (explicit or implicit - if any) and how that impacts linearity of gas cost of processing this transaction at the increase of the depth of the names.

Reply:
It's now handled in implementation - we don't create intermediate nodes in aens tree. Processing cost of the list of subnames with pointers should be linear to the tx size.